Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/ContextMtl.mm

```diff
@@ -2278,8 +2278,21 @@ GLint GetOwnershipIdentity(const egl::AttributeMap &attribs)
         ANGLE_TRY(setupDrawImpl(context, mode, firstVertex, vertexOrIndexCount, instances,
                                 indexTypeOrNone, indices, xfbPass));
         // Setup with flushed state should either produce a working encoder or fail with an error
-        // result.
-        ASSERT(mRenderEncoder.valid());
+        // result. if that's not the case, something's gone seriously wrong. Try to
+        // recover from the error by bailing out of the draw call, and finishing the command buffer. This will result in an unfinished / corrupted draw,
+        // but will avoid a browser/GPU process crash.
+        if(ANGLE_UNLIKELY(!mRenderEncoder.valid() || !mRenderEncoder.hasPipelineState()))
+        {
+            //Completely flush the command buffer, waiting synchronously.
+            flushCommandBuffer(mtl::WaitUntilFinished);
+            //Invalidate all state
+            invalidateState(context);
+            //Return fail, and drop the draw call. This is
+            //a worst case scenario. If in a debug roots situation,
+            //we should try to catch the call stack.
+            ASSERT(!"Draw call is unusable");
+            return angle::Result::Stop;
```

Comment:
Does this result in ANGLE issuing an error message that would alert a user to submit an issue?

It would be beneficial to get a handle on which websites are causing this issue.

Reply:
This won't. Grepping around a bit, consider:

`ANGLE_CHECK(context, false, "Draw call is unusable", GL_INVALID_OPERATION);`

Similar pattern seen in Program.cpp.